kratos/includes/dof.h

```diff
@@ -529,39 +527,37 @@ class Dof{
 
     void save(Serializer& rSerializer) const 
     {
-        rSerializer.save("Is Fixed", static_cast<bool>(mIsFixed));
-        rSerializer.save("Equation Id", static_cast<EquationIdType>(mEquationId));
-        rSerializer.save("Nodal Data", mpNodalData);
-        rSerializer.save("Variable", mpVariable->Name());
-        rSerializer.save("Reaction", mpReaction->Name());
-        rSerializer.save("Variable Type", static_cast<int>(mVariableType));
-        rSerializer.save("Reaction Type", static_cast<int>(mReactionType));
+        rSerializer.save("IsFixed", static_cast<bool>(mIsFixed));
+        rSerializer.save("EquationId", static_cast<EquationIdType>(mEquationId));
+        rSerializer.save("NodalData", mpNodalData);
+        rSerializer.save("VariableType", static_cast<int>(mVariableType));
+        rSerializer.save("ReactionType", static_cast<int>(mReactionType));
+        rSerializer.save("Index", static_cast<int>(mIndex));
+
     }
 
     void load(Serializer& rSerializer) 
     {
         std::string name;
         bool is_fixed;
-        rSerializer.load("Is Fixed", is_fixed);
+        rSerializer.load("IsFixed", is_fixed);
         mIsFixed=is_fixed;
         EquationIdType equation_id;
-        rSerializer.load("Equation Id", equation_id);
+        rSerializer.load("EquationId", equation_id);
         mEquationId = equation_id;
-        rSerializer.load("Nodal Data", mpNodalData);
-        rSerializer.load("Variable", name);
-        mpVariable=KratosComponents<VariableData>::pGet(name);
-        rSerializer.load("Reaction", name);
-        if(name == "NONE")
-            mpReaction = &msNone;
-        else
-            mpReaction=KratosComponents<VariableData>::pGet(name);
+        rSerializer.load("NodalData", mpNodalData);
+
         int variable_type;
         int reaction_type;
-        rSerializer.load("Variable Type", variable_type);
-        rSerializer.load("Reaction Type", reaction_type);
+        rSerializer.load("VariableType", variable_type);
+        rSerializer.load("ReactionType", reaction_type);
 
         mVariableType = variable_type;
         mReactionType = reaction_type;
+
+        int index;
+        rSerializer.load("Index", index);
+        mIndex = index;
```

Comment:
why is the direct load not possible here?

Reply:
Because I wasn't sure if it takes the correct type between all overloaded ones. So I choose it explicitly.